lib/haml/plugin.rb

```diff
@@ -4,6 +4,11 @@ module Haml
 
   # This module makes Haml work with Rails using the template handler API.
   class Plugin
+    class << self
+      attr_accessor :annotate_rendered_view_with_filenames
+    end
+    self.annotate_rendered_view_with_filenames = false
```

Comment:
Instead of creating Haml's own config, how about checking `ActionView::Base.try(:annotate_rendered_view_with_filenames)`? I think it's what Rails checks for .erb files too, and I don't think we need to support it with Rails versions that don't support it for .erb files.

Reply:
Oh sorry I overlooked what you wrote in `lib/haml/railtie.rb`. But at least I want this `Haml::Plugin.annotate_rendered_view_with_filenames` to either be private or not exist for maintainability of Haml and consistency with .erb files.

Reply:
Could do - I did it this way because it seemed like it would be useful even outside of Rails. But if it'll help get this merged, I'll update it.

Reply:
I think I agree with @jdelStrother here... as really Haml::Plugin is for ActionView, and the railtie configures it with the Rails App settings. I mean, at this point the whole abstraction of the Plugin is kind of weird anyway and probably needs revisiting in later versions.

The main advantage of doing this as it's 'own setting' is it's just much more testable too.